clients/src/main/java/org/apache/kafka/clients/NetworkClient.java

```diff
@@ -363,8 +359,10 @@ public Node leastLoadedNode(long now) {
         List<Node> nodes = this.metadataUpdater.fetchNodes();
         int inflight = Integer.MAX_VALUE;
         Node found = null;
+
+        int offset = this.randOffset.nextInt(nodes.size());
         for (int i = 0; i < nodes.size(); i++) {
-            int idx = Utils.abs((this.nodeIndexOffset + i) % nodes.size());
+            int idx = (offset + i) % nodes.size();
```

Comment:
I could be wrong, but that seems equivalent to what I have above. If we're really concerned about this overflow, the easiest thing would be to do the addition using longs and convert back to int after doing the modulus. Seems like overkill though since no one is likely to scale Kafka past a billion nodes. And the impact of overflow is benign anyway when you have so many nodes to choose from.